utility/unit_of_work/module.go

```diff
@@ -0,0 +1,209 @@
+package unit_of_work
+
+import (
+	coreTypes "github.com/pokt-network/pocket/shared/core/types"
+	"github.com/pokt-network/pocket/shared/modules"
+	"github.com/pokt-network/pocket/shared/modules/base_modules"
+	utilTypes "github.com/pokt-network/pocket/utility/types"
+)
+
+const (
+	leaderUtilityUOWModuleName  = "leader_utility_UOW"
+	replicaUtilityUOWModuleName = "replica_utility_UOW"
+)
+
+var _ modules.UtilityUnitOfWork = &baseUtilityUnitOfWork{}
+
+type baseUtilityUnitOfWork struct {
+	base_modules.IntegratableModule
+
+	logger *modules.Logger
+
+	height int64
+
+	persistenceReadContext modules.PersistenceReadContext
```

Comment:
Can you please add a comment explaining why we have both with a potential example use case.